internal/planmodifiers/caseinsensitive.go

```diff
@@ -0,0 +1,35 @@
+package planmodifiers
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func CaseInsensitiveString() planmodifier.String {
+	return &caseInsensitivePlanModifier{}
+}
+
+type caseInsensitivePlanModifier struct{}
+
+func (d *caseInsensitivePlanModifier) PlanModifyString(ctx context.Context, request planmodifier.StringRequest, response *planmodifier.StringResponse) {
+	oldValue := request.StateValue.ValueString()
+	newValue := request.PlanValue.ValueString()
+
+	result := oldValue
+	if !strings.EqualFold(strings.ToLower(newValue), strings.ToLower(oldValue)) {
```

Comment:
Nit: EqualFold ignores case, so why convert the args to lower-case?

Reply:
Right fixed